### DISCUSSION TITLE
Support Ruby as a linter

### DISCUSSION THREAD

**Discussion by bartekb** [zed-industries/zed#26841] [Feature Requests]
### Check for existing issues

- [X] Completed

### Describe the feature

Is it possible to use `ruby -c` as a linter? For some reasons, I can't and don't want to use slow `ruby-lsp` or the very slow `rubocop`. I definitely prefer syntax checking via the Ruby binary. This works in VSCode and Vim/Neovim, but it seems not to be possible in Zed.

### Environment

Zed: v0.165.4 (Zed)
OS: macOS 15.2.0
Memory: 36 GiB
Architecture: aarch64

**Comment by jansol** [comment]
Does that make ruby act as a language server according to the language server protocol? If not, it requires a lot of manual work to support.

**Comment by bartekb** [comment]
Not like LSP, unfortunately. It's just the use of an external interpreter to point out syntax errors in the code, similar to external formatters when formatting through an LSP tool is not available.

**Comment by notpeter** [comment]
You could implement this as a [Task](https://zed.dev/docs/tasks) which when successful had no output and when there were errors would include links to the files/lines in question.

There isn't currently a way for external tools to provide diagnostic feedback into zed and integrated into the editor buffer or diagnostics, except via LSP.  It's not impossible, but not something on our roadmap.

When you say this works in VSCode, what exactly do you mean? Is this built in? Or is there an extension that will spawn `ruby -c` on save, process the output and then annotate your file?
